src/web/rustdoc.rs

```diff
@@ -253,6 +253,12 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
     let metrics = extension!(req, Metrics).clone();
     let mut rendering_time = RenderingTimesRecorder::new(&metrics.rustdoc_rendering_times);
 
+    // Pages generated by Rustdoc are not ready to be served with a CSP yet.
+    req.extensions
+        .get_mut::<Csp>()
+        .expect("missing CSP")
+        .suppress(true);
```

Comment:
This is shared between all threads, isn't it? Won't this surpress the CSP policy for all requests? Or is it per-request because you used `link_before`?

Reply:
Well, all the extensions [are added in each request](https://github.com/rust-lang/docs.rs/blob/master/src/web/extensions.rs). Most of the others are just cloned `Arc<_>`s, instead here it's a fresh instance every time.